addons/fleet/models/fleet_vehicle.py

```diff
@@ -299,6 +299,9 @@ def create(self, vals_list):
         return vehicles
 
     def write(self, vals):
+        if 'odometer' in vals:
+            old_odometers = self.mapped('odometer')
```

Comment:
@Flotchet we should move at the end of the function we check `if 'odometer' in vals:` twice